queue/queue.go

```diff
@@ -35,11 +38,31 @@ type MetricsSource interface {
 }
 
 func New() (*Queue, error) {
-	return &Queue{}, nil
+	eventQueue := memqueue.NewQueue(logp.L(), memqueue.Settings{
+		Events: 1024,
+		// The event count and timeout for queue flushes is hard-coded to a placeholder
+		// for now, since that's what the existing Beats API understands. The plan is
+		// for these parameters to instead be controlled by the output as it reads
+		// the queue.
+		FlushMinEvents: 256,
+		FlushTimeout:   5 * time.Millisecond,
+	})
+	producer := eventQueue.Producer(
+		beatsqueue.ProducerConfig{})
+	if producer == nil {
+		fmt.Printf("\033[94mwhy is the queue producer nil\033[0m\n")
+	}
+	return &Queue{eventQueue: eventQueue, producer: producer}, nil
 }
 
 func (queue *Queue) Publish(event *api.Event) error {
-	return fmt.Errorf("couldn't publish: Queue.Publish is not implemented")
+	if queue.producer == nil {
+		fmt.Printf("\033[94mWAT\033[0m\n")
```

Comment:
Same, I think we should return an error.

Reply:
(same as above, if the queue itself is initialized then producer is guaranteed to be non-nil)